module/VuFind/src/VuFind/View/Helper/Root/Flashmessages.php

```diff
@@ -111,9 +111,18 @@ public function __invoke()
                         $helper = (isset($msg['html']) && $msg['html'])
                             ? false : 'escapeHtml';
                     }
-                    $helper = $helper
-                        ? $this->getView()->plugin($helper) : false;
+                    $helper = $helper ? $this->getView()->plugin($helper) : false;
                     $tokens = $msg['tokens'] ?? [];
+                    if ($tokens && $mode = ($msg['translateTokens'] ?? false)) {
+                        // Escape translated tokens unless html is requested or the
+                        // main message is translated:
+                        $translator = 'html' === $mode || empty($msg['html'])
```

Comment:
This line of code has me really confused, though if tests are passing, maybe I'm just totally misunderstanding what I'm looking at... but why are we comparing `$mode` (which appears to be a boolean) with the string `'html'`? Why is `$msg['html']` being empty (as opposed to non-empty) causing us to go into non-escaping translation mode? Am I misreading the operator associativity or something, or is something wrong here?

Reply:
It was a bit messy, and coupling translation and escaping of tokens perhaps wasn't the best idea ever. I've now decoupled them and rewritten the part. Hope it makes more sense now!